cmd/operator-sdk/bundle/add.go

```diff
@@ -0,0 +1,227 @@
+func newAddCmd() *cobra.Command {
+	c := &bundleIndexCmd{}
+	cmd := &cobra.Command{
+		Use:   "add",
+		Short: "Add an operator bundle image to an operator index image",
+		Long: fmt.Sprintf(`The 'operator-sdk bundle add' command will add an operator bundle image to an
+existing operator index image, or create an index image.
+
+This command downloads and shells out to the 'opm' binary under the hood. The
+version downloaded by 'bundle add' is: %s
+
+Bundle images being passed to 'bundle add' must be present remotely, and access
+to the remote repository should be enabled in the command line environment.
+
+More information on operator index images:
+https://github.com/openshift/enhancements/blob/master/enhancements/olm/operator-registry.md
+More information on 'opm':
+https://github.com/operator-framework/operator-registry/blob/master/docs/design/opm-tooling.md
+`,
+			registryVersion),
+		Example: `The following invocation will create a new test-operator bundle index image:
+
+  $ operator-sdk bundle add quay.io/example/test-operator:v0.1.0 \
+      --to-index quay.io/example/test-operator-index:v0.1.0
+
+The following invocation will add a test-operator bundle image to an existing
+index image at version v0.1.0, creating a new index image at version v0.2.0:
+
+  $ operator-sdk bundle add quay.io/example/test-operator:v0.2.0 \
+      --from-index quay.io/example/test-operator-index:v0.1.0 \
+      --to-index quay.io/example/test-operator-index:v0.2.0
+`,
+		RunE: func(cmd *cobra.Command, args []string) error {
+			if len(args) != 1 {
+				return errors.New("a comma-separated list of bundle image tags is a required argument, " +
+					"ex. 'example.com/test-operator:v0.1.0,example.com/test-operator:v0.2.0'")
+			}
+			c.bundlesImages = args[0]
+
+			if err := c.validate(); err != nil {
+				return fmt.Errorf("error validating args: %v", err)
+			}
+
+			binaryPath := filepath.Join("bin", "opm")
+			if runtime.GOOS == "windows" {
+				binaryPath += ".exe"
+			}
+			if err := c.buildOPM(binaryPath); err != nil {
+				log.Fatalf("Error building image builder: %v", err)
+			}
+			if err := c.runOPMIndexAdd(binaryPath); err != nil {
+				log.Fatalf("Error buildingindex image: %v", err)
+			}
+
+			return nil
+		},
+	}
+
+	cmd.Flags().StringVarP(&c.fromIndex, "from-index", "f", "", "Previous index to build new index image from")
+	cmd.Flags().StringVarP(&c.toIndex, "to-index", "t", "", "Tag for new index image being built")
+	if err := cmd.MarkFlagRequired("to-index"); err != nil {
+		panic(err)
+	}
+	cmd.Flags().StringVar(&c.dockerfileName, "dockerfile-name", "",
+		"Name of the Dockerfile to generate if --generate-only is set. Default is 'Dockerfile'")
+	cmd.Flags().StringVar(&c.imageBuilder, "image-builder", "docker",
+		"Tool to build container images. One of: [docker, podman]")
+	cmd.Flags().BoolVarP(&c.generateOnly, "generate-only", "g", false,
+		"Generate metadata and a Dockerfile on disk without building the bundle image")
```

Comment:
Should this comment be `Generate the underlying database and a Dockerfile without building the index container image`?

Reply:
If that's what `opm index add --generate` is actually doing then yes. The help text only says it generates a Dockerfile:
```console
$ opm index add --help
      ...
      --generate                if enabled, just creates the dockerfile and saves it to local disk
```

Reply:
Also I forgot to update this text after copying from `create`, whoops.

Reply:
yeah, it does. we need to update this help text to make that more clear.